spec/double_entry/reporting_spec.rb

```diff
@@ -42,4 +42,160 @@
+          transfers.define(:from => :savings, :to => :cash,    :code => :spend)
+          transfers.define(:from => :cash,    :to => :savings, :code => :save)
+          transfers.define(:from => :cash,    :to => :credit,  :code => :bill)
+        end
+      end
+
+      cash    = DoubleEntry.account(:cash)
+      savings = DoubleEntry.account(:savings)
+      credit  = DoubleEntry.account(:credit)
+      DoubleEntry.transfer(Money.new(30_00), :from => cash,    :to => savings, :code => :save)
+      DoubleEntry.transfer(Money.new(10_00), :from => cash,    :to => savings, :code => :save)
+      DoubleEntry.transfer(Money.new(20_00), :from => cash,    :to => credit,  :code => :bill)
+      DoubleEntry.transfer(Money.new(10_00), :from => savings, :to => cash,    :code => :spend)
+
+      first_transfer         = DoubleEntry::Line.first
+      last_transfer          = DoubleEntry::Line.last
+      DoubleEntry::LineMetadata.create!(:line => first_transfer,         :key => :reason,   :value => :payday)
+      DoubleEntry::LineMetadata.create!(:line => first_transfer.partner, :key => :reason,   :value => :payday)
+      DoubleEntry::LineMetadata.create!(:line => last_transfer,          :key => :category, :value => :entertainment)
+      DoubleEntry::LineMetadata.create!(:line => last_transfer.partner,  :key => :category, :value => :entertainment)
+    end
+
+    after do
+      # restore "helpful" predefined config
+      DoubleEntry.configuration.accounts = @config_accounts
+      DoubleEntry.configuration.transfers = @config_transfers
+    end
+
+    describe 'filter solely on transaction identifiers and time' do
+      let(:function) { :sum }
+      let(:account) { :savings }
+      let(:code) { :save }
+      let(:range) { DoubleEntry::Reporting::MonthRange.current }
+
+      subject(:aggregate) do
+        DoubleEntry::Reporting.aggregate(
+          function, account, code,
+          { :range => range }
+        )
+      end
+
+      specify 'Total attempted to save' do
+        expect(aggregate).to eq(Money.new(40_00))
+      end
+    end
+
+    describe 'filter by named scope that does not take arguments' do
+      let(:function) { :sum }
+      let(:account) { :savings }
+      let(:code) { :save }
+      let(:range) { DoubleEntry::Reporting::MonthRange.current }
+
+      subject(:aggregate) do
+        DoubleEntry::Reporting.aggregate(
+          function, account, code,
+          {
+            :range  => range,
+            :filter => [
+              :scope => {
+                :name => :ten_dollar_transfers
+              }
+            ]
+          }
+        )
+      end
+
+      before do
+        DoubleEntry::Line.class_eval do
+          scope :ten_dollar_transfers, -> do
+            where(:amount => Money.new(10_00).fractional)
+          end
+        end
+      end
+
+      specify 'Total amount of $10 transfers attempted to save' do
+        expect(aggregate).to eq(Money.new(10_00))
```

Comment:
There's only one transfer that falls into this scope. If there were a bug where only the first item was returned rather than all items, then this test would not pick that up. I'd prefer to have more than one in the setup.